core/src/main/java/org/apache/iceberg/TableUtil.java

```diff
@@ -30,6 +30,9 @@ public static int formatVersion(Table table) {
     if (table instanceof SerializableTable) {
       SerializableTable serializableTable = (SerializableTable) table;
       return serializableTable.formatVersion();
+    } else if (table instanceof PositionDeletesTable) {
+      PositionDeletesTable positionDeletesTable = (PositionDeletesTable) table;
```

Comment:
Can't we replace this with `BaseMetadataTable` and then we don't end up being too specific here.